velodyne_driver/src/driver/driver.cc

```diff
@@ -62,7 +62,12 @@ VelodyneDriver::VelodyneDriver(ros::NodeHandle node,
   private_nh.param("model", config_.model, std::string("64E"));
   double packet_rate;                   // packet frequency (Hz)
   std::string model_full_name;
-  if ((config_.model == "64E_S2") || 
+  if ((config_.model == "VLS128") )
+  {
+    packet_rate = 12507;     //https://github.com/velodynevls/velodyne_vls/blob/master/velodyne_driver/src/driver/driver.cc
```

Comment:
We took this from the repository, but are currently evaluating it because  with the information of the manual we get only half of this value. We have already made some measurements using the first value and they look good but we will test the other value in the next couple of weeks.

Reply:
I would guess that this value is for when dual-return mode is active. By default, most Velodyne sensors are configured with either "strongest return" or "closest return" which provides only half the data. Not all sensors provide dual-return mode and this driver is currently not set up to support it so I recommend setting this to half the current size if that is what you're seeing from your sensor.

Reply:
Yea, that may be true. We think the value was calculated using wireshark or something similar, but provably they forgot to account for the positioning messages that the sensor sends. I have replaced it by the value calculated in the manual.

Reply:
LGTM!